### ISSUE TITLE
Can't play a file with VoiceClient::playFile()

### ISSUE THREAD

**Issue by ruppertphil** [discord-php/DiscordPHP#572]
**Environment**
- PHP Version:
    - 8.0.9
- DiscordPHP Versions:
    - * dev-develop
- Ubuntu 21.04
- FFmpeg

**Describe the bug**
The bot joins my voice channel but plays no sound. There is no error and it looks like it can find the mp3 file. Also tried a .wav file.

**Code**
```php
$discord->joinVoiceChannel($voiceChannel, false, false)->done(function(VoiceClient $voiceClient){
    $voiceClient->playFile("dt.mp3")->otherwise(function($e){ echo "ERROR: ".$e->getMessage(); });
});
```

**Expected behavior**
Join a voice channel and play a sound.

**Comment by oliverschloebe** [comment]
Afaik the voice client needs `ext-sodium` for voice encryption which your system may be missing.

Also have you tried using an absolute path with the MP3 file, e.g. `dirname(__DIR__) . "/dt.mp3"`?

**Comment by ruppertphil** [comment]
Sodium for PHP 8 is already installed and i also tried it with the absolute path.

**Comment by crustamet** [comment]
I am having the same issue, i have installed the ffmpeg binary on my host, installed Sodium extension for PHP

everything works perfectly but no sound from the bot...

**Comment by crustamet** [comment]
i think im on to something here

while using this

	$discordCommands->joinVoiceChannel($channel)->then(function (VoiceClient $vc) 
	{
		echo 'test';
		var_dump($vc);die();
		$vc->playFile('cedry2k.mp3');
	}, function ($e) {
		echo "There was an error joining the voice channel: {$e->getMessage()}"; 
	})->otherwise(function($e){ 
		echo "ERROR: ".$e->getMessage(); 
	});

i found out that it does not enter the **then** method to play the file it does not reach that line, but no error whatsoever also

**Comment by crustamet** [comment]
after alot of die() line by line i found that the socket client is emmiting errors

**Unable to send packet: stream_socket_sendto(): Operation not permitted** at line 430 on Voice Client php file

     $client->on('error', function ($e) {
          $this->emit('udp-error', [$e]);
     });

i do not know what to do now i know it gives error that is why is not playing the audio file

**Comment by crustamet** [comment]
It says in the output file that the bot connects to port 50006 

isn't this supposed to be 5 006 instead of 50 006 ?

**Comment by degoya** [comment]
did you get any progress on playing the sound? i've also got the problem that i can't play any sound :(

**Comment by crustamet** [comment]
Yes i have the icon too but that icon means the bot is deafened means he can not hear anything but can speak if the playfile works.
I am still waiting for a dev to respond to this i have no idea what is the problem

**Comment by davidcole1340** [comment]
I can't reproduce the problem, sound plays fine for me. By default the bot is deafened as it saves bandwidth if you aren't using the received audio, but that doesn't affect your ability to 'speak'.

Make sure you have `ext-sodium` installed and FFmpeg, as well as running Linux, macOS or Windows on an AMD64 system. Try running DCA from the `bin` folder to see if it executes properly.

Try add an error handler to the voice client join promise:

```php
$discord->joinVoiceChannel($channel)->then(function (VoiceClient $client) {
    $client->on('error', function ($e) {
        echo "client error\n";
        var_dump($e);
    });

    $client->on('stderr', function ($e) {
        echo "DCA stderr\n";
        var_dump($e);
    });

    $client->playFile('./filename.mp3')->then(function () {
        echo "done\n";
    }, function ($e) {
        echo "failed to play song:\n";
        var_dump($e);
    }); 
}, function ($e) {
    echo "failed to join voice channel:\n";
    var_dump($e);
});
```

**Comment by ruppertphil** [comment]
Thx for all your replies!
I've tried it now with all the error handlers, but no error in the output at all. 

`[2021-09-29T00:54:26.440443+00:00] DiscordPHP.INFO: voice client is ready [] []
[2021-09-29T00:54:26.440620+00:00] DiscordPHP.INFO: set voice client bitrate {"bitrate":64000} []
`
That`s all i get.

**Comment by davidcole1340** [comment]
Can you run `php -i` and `composer show` and paste both outputs here please? Note that the first output will contain your environment variables so make sure to remove anything sensitive.

Can you let me know what OS you are running as well? If it's unix based please run `uname -a` as well.

**Comment by ruppertphil** [comment]
Php -i [here](https://gist.github.com/ruppertphil/b2caad7b08dba70596ea84a47956500f)
Composer [here](https://gist.github.com/ruppertphil/4bec8039093b8abd415026ded58d8a83)

I'm running Ubuntu 21.04 
`Linux vps-c47b**** 5.11.0-31-generic #33-Ubuntu SMP Wed Aug 11 13:19:04 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`

Could AMD64 be a problem here? It's a rented VPS.

**Comment by davidcole1340** [comment]
You are on a `x86_64` platform so that's alright. I noticed you are using an older version of DiscordPHP, could you try update to the latest `master` branch? The `develop` branch is no longer being used at the moment so change your version to `dev-master`.

**Comment by davidcole1340** [comment]
Also, try running `./vendor/team-reflex/discord-php/bin/dca-v0.1.0-linux-amd64` in the root folder of your bot and see what the output is.

**Comment by ruppertphil** [comment]
Okay i've updated Discord-PHP and nothing changed.
**But**, i had some problems running **dca-v0.1.0-linux-amd64** due to **missing permissions**. I've added execution permissions on that file and now the bot can play sounds.

Thx! Let's see if that's a fix for the other devs :)

**Comment by davidcole1340** [comment]
Don't know why that would happen as Git preserves executable permissions on files, however it might be something with Composer.

`stderr` wasn't actually working so I've fixed that as well, hopefully this is easier to diagnose next time.

**Comment by crustamet** [comment]
Here
https://github.com/discord-php/DiscordPHP/blob/c635bcb5bd554284f7e77f088b23081ea62a1edd/src/Discord/Voice/VoiceClient.php#L431

On this line instead of udp-error it should emit an error because udp-error is not treated anywhere in the project.

When i changed that now i receive the errors i was talking about

	(
		[message:protected] => Unable to send packet: stream_socket_sendto(): Operation not permitted
		[string:Exception:private] => 
		[code:protected] => 0
		[file:protected] => ./vendor/react/datagram/src/Buffer.php
		[line:protected] => 115
		[trace:Exception:private] => Array
			(
				[0] => Array
					(
						[file] => ./vendor/react/datagram/src/Buffer.php
						[line] => 43
						[function] => handleWrite
						[class] => React\Datagram\Buffer
						[type] => ->
						[args] => Array
							(
								[0] =>    ¹                                                                 
								[1] => 
							)

					)

				[1] => Array
					(
						[file] => ./vendor/react/event-loop/src/StreamSelectLoop.php
						[line] => 253
						[function] => onWritable
						[class] => React\Datagram\Buffer
						[type] => ->
						[args] => Array
							(
								[0] => Resource id #261
							)

					)

				[2] => Array
					(
						[file] => ./vendor/react/event-loop/src/StreamSelectLoop.php
						[line] => 212
						[function] => waitForStreamActivity
						[class] => React\EventLoop\StreamSelectLoop
						[type] => ->
						[args] => Array
							(
								[0] => 3588501
							)

					)

				[3] => Array
					(
						[file] => ./vendor/team-reflex/discord-php/src/Discord/Discord.php
						[line] => 1368
						[function] => run
						[class] => React\EventLoop\StreamSelectLoop
						[type] => ->
						[args] => Array
							(
							)

					)

				[4] => Array
					(
						[file] => ./projects/discord/mybot.php
						[line] => 2462
						[function] => run
						[class] => Discord\Discord
						[type] => ->
						[args] => Array
							(
							)

					)

			)

		[previous:Exception:private] => 
	)
	Exception Object
	(
		[message:protected] => Invalid message
		[string:Exception:private] => 
		[code:protected] => 0
		[file:protected] => ./vendor/react/datagram/src/Socket.php
		[line:protected] => 123
		[trace:Exception:private] => Array
			(
				[0] => Array
					(
						[file] => ./vendor/react/datagram/src/Socket.php
						[line] => 67
						[function] => handleReceive
						[class] => React\Datagram\Socket
						[type] => ->
						[args] => Array
							(
								[0] => 
							)

					)

				[1] => Array
					(
						[file] => ./vendor/react/event-loop/src/StreamSelectLoop.php
						[line] => 245
						[function] => onReceive
						[class] => React\Datagram\Socket
						[type] => ->
						[args] => Array
							(
								[0] => Resource id #261
							)

					)

				[2] => Array
					(
						[file] => ./vendor/react/event-loop/src/StreamSelectLoop.php
						[line] => 212
						[function] => waitForStreamActivity
						[class] => React\EventLoop\StreamSelectLoop
						[type] => ->
						[args] => Array
							(
								[0] => 3588206
							)

					)

				[3] => Array
					(
						[file] => ./vendor/team-reflex/discord-php/src/Discord/Discord.php
						[line] => 1368
						[function] => run
						[class] => React\EventLoop\StreamSelectLoop
						[type] => ->
						[args] => Array
							(
							)

					)

				[4] => Array
					(
						[file] => ./projects/discord/mybot.php
						[line] => 2462
						[function] => run
						[class] => Discord\Discord
						[type] => ->
						[args] => Array
							(
							)

					)

			)

		[previous:Exception:private] => 
	)

**Comment by degoya** [comment]
changing the permissions on the file ./vendor/team-reflex/discord-php/bin/dca-v0.1.0-linux-amd64  the sounds play for me too, maybe this should be documented somewhere.

now when my sound plays, how i can stop it to play another sound?
